packages/ckb-sdk-utils/src/address/index.ts

```diff
@@ -8,63 +8,103 @@ export enum AddressPrefix {
 }
 
 export type CodeHashIndex = '0x00' | string
 
 export interface AddressOptions {
   prefix: AddressPrefix
   type: AddressType
-  codeHashIndex: CodeHashIndex
+  codeHashOrCodeHashIndex: CodeHashIndex | CKBComponents.Hash256
 }
 
 export const defaultAddressOptions = {
   prefix: AddressPrefix.Testnet,
   type: AddressType.HashIdx,
-  codeHashIndex: '0x00',
+  codeHashOrCodeHashIndex: '0x00',
 }
 
 /**
+ * @function toAddressPayload
  * @description payload = type(01) | code hash index(00) | blake160-formatted pubkey
  * @see https://github.com/nervosnetwork/ckb/wiki/Common-Address-Format
+ * @param {string | Uint8Array} arg, use as the identifier of an address, usually the public key hash is used.
+ * @param {string} type, used to indicate which format is adopted to compose the address.
+ * @param {string} codeHashOrCodeHashIndex, the referenced code hash or code hash index the address binds to
  */
 export const toAddressPayload = (
-  publicKeyHash: string | Uint8Array,
+  arg: string | Uint8Array,
   type: AddressType = AddressType.HashIdx,
-  params: CodeHashIndex = '0x00'
+  codeHashOrCodeHashIndex: CodeHashIndex | CKBComponents.Hash256 = '0x00'
 ): Uint8Array => {
-  if (typeof publicKeyHash === 'string') {
-    if (!publicKeyHash.startsWith('0x')) {
-      throw new HexStringShouldStartWith0x(publicKeyHash)
+  if (typeof arg === 'string') {
+    if (!arg.startsWith('0x')) {
+      throw new HexStringShouldStartWith0x(arg)
     }
-    return new Uint8Array([...hexToBytes(type), ...hexToBytes(params), ...hexToBytes(publicKeyHash)])
+    return new Uint8Array([...hexToBytes(type), ...hexToBytes(codeHashOrCodeHashIndex), ...hexToBytes(arg)])
   }
-  return new Uint8Array([...hexToBytes(type), ...hexToBytes(params), ...publicKeyHash])
+  return new Uint8Array([...hexToBytes(type), ...hexToBytes(codeHashOrCodeHashIndex), ...arg])
 }
 
+/**
+ * @name bech32Address
+ * @description generate the address by bech32 algorithm
+ * @param arg, used as the identifier of an address, usually the public key hash is used.
+ * @param {string} prefix, the prefix precedes the address.
+ * @param {string} type, used to indicate which format is adopted to compose the address.
+ * @param {string} codeHashOrCodeHashIndex, the referenced code hash or code hash index the address binds to.
+ */
 export const bech32Address = (
-  publicKeyHash: Uint8Array | string,
+  arg: Uint8Array | string,
   {
     prefix = AddressPrefix.Testnet,
     type = AddressType.HashIdx,
-    codeHashIndex = '0x00',
+    codeHashOrCodeHashIndex = '0x00',
   }: AddressOptions = defaultAddressOptions
-) => bech32.encode(prefix, bech32.toWords(toAddressPayload(publicKeyHash, type, codeHashIndex)))
+) => bech32.encode(prefix, bech32.toWords(toAddressPayload(arg, type, codeHashOrCodeHashIndex)))
+
+/**
+ * @name fullPayloadToAddress
+ * @description generate the address with payload in full version format.
+ * @param {string} arg, used as the identifier of an address.
```

Comment:
Although it's a single value of bytes I believe we still call it `args`, instead of `arg`.

Reply:
updated in https://github.com/nervosnetwork/ckb-sdk-js/pull/361/commits/2217205f83c06ca8df908d0c282df230494612bb